pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -175,7 +175,7 @@ public void init(SegmentGeneratorConfig config, SegmentCreationDataSource dataSo
 
     // Initialize index creation
     _segmentIndexCreationInfo = new SegmentIndexCreationInfo();
-    _indexCreationInfoMap = new HashMap<>();
+    _indexCreationInfoMap = new LinkedHashMap<>();
```

Comment:
Thanks for working on this!

I have some questions:
1. Is the only goal to fix flaky tests?
2. HashMap and LinkedHashMap have different performance on different operations. Changing HashMap to LinkedHashMap may slow down the index creation. This may be a problem it the index creation is slowed down too much. Did you try to compare the performance and can you share the results if you did?